File: Baekjoon/Baekjoon10164.py
# Baekjoon10164_PathOnGrid.py
# https://www.acmicpc.net/problem/10164
def GetNumberOfPath( row, col ):
	dp = [ [ 0 for _ in range( col ) ] for _ in range( row ) ]

	for i in range( row ):
		dp[i][0] = 1

	for i in range( col ):
		dp[0][i] = 1

	for i in range( 1, row ):
		for j in range( 1, col ):
			dp[i][j] = dp[i - 1][j] + dp[i][j - 1]

	return dp[row - 1][col - 1]

n, m, k = map( int, input().split() )

midRow, midCol = 0, 0
num, isFind = 1, False
for i in range( n ):
	for j in range( m ):
		if ( num == k ):
			midRow = i
			midCol = j
			isFind = True
			break
		num += 1

	if ( isFind ):
		break

print( GetNumberOfPath( midRow + 1, midCol + 1 ) * GetNumberOfPath( n - midRow, m - midCol ) )

File: Baekjoon/Baekjoon10164.cpp
// Baekjoon10164_PathOnGrid.cpp
#include <iostream>

using namespace std;

int main(void) {
	int row, col, num;
	cin >> row >> col >> num;

	bool isFind = false;
	int targetRow = 0, targetCol = 0, curr = 1;
	for ( int i = 0; i < row; i++ ) {
		for ( int j = 0; j < col; j++ ) {
			if ( curr == num ) {
				targetRow = i, targetCol = j;
				isFind = true;
				break;
			}
			curr++;
		}

		if ( isFind )
			break;
	}

	int dp[15][15] = { 0, };
	for ( int i = 0; i < row; i++ )
		dp[i][0] = 1;
	for ( int j = 0; j < col; j++ )
		dp[0][j] = 1;

	for ( int i = 1; i <= targetRow; i++ ) {
		for ( int j = 1; j <= targetCol; j++ ) {
			dp[i][j] = dp[i - 1][j] + dp[i][j - 1];
		}
	}

	int targetPath = dp[targetRow][targetCol];
	for ( int i = targetRow; i < row; i++ )
		dp[i][targetCol] = 1;
	for ( int j = targetCol; j < col; j++ )
		dp[targetRow][j] = 1;

	for ( int i = targetRow + 1; i < row; i++ )
		for ( int j = targetCol + 1; j < col; j++ )
			dp[i][j] = dp[i - 1][j] + dp[i][j - 1];

	cout << dp[row - 1][col - 1] * targetPath << '\n';

	return 0;
}